lib/rules/no-promise-without-done-fail.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+/**
+ * Determines whether a function call is a Promise.property() call
+ * @param {ASTNode} node A CallExpression node
+ * @returns {boolean} `true` if the call is a Promise.property() call
+ */
+function hasIdentifier (node, property) {
+  return node.callee &&
+    node.callee.type === 'MemberExpression' &&
+    node.callee.property.type === 'Identifier' &&
+    node.callee.property.name === property
+}
+
+function insideReturnStatement (node) {
+  return node.parent.type === 'ReturnStatement'
+}
+
+function hasDoneFailInReject (node, asyncParam) {
+  var reject = node.arguments[1]
+  return reject &&
+    reject.type === 'MemberExpression' &&
+    reject.object.type === 'Identifier' && reject.object.name === asyncParam &&
+    reject.property.type === 'Identifier' && reject.property.name === 'fail'
+}
+
+function isPromiseThenWithoutCatch (node, asyncParam) {
+  return hasIdentifier(node, 'then') &&
+    !hasIdentifier(node.parent.parent, 'then') &&
+    !insideReturnStatement(node) &&
```

Comment:
why do you need this condition?

Reply:
If you have code that you need to test asynchronously, you can declare that you receive a done callback, *return a Promise*, or use the async keyword (https://jasmine.github.io/api/edge/global.html#implementationCallback)

Reply:
Looking at the tests you provided `!insideReturnStatement(node)` doesn't catch any cases. If you comment this condition out, all of the tests pass anyways. So either add a test case that is caught by this condition or remove it. Right now this condition is unnecessary.